tests/ludwig/encoders/test_text_encoders.py

```diff
@@ -292,3 +289,23 @@ def test_tfidf_encoder(vocab_size: int):
     inputs = torch.randint(2, (batch_size, sequence_length)).to(DEVICE)
     outputs = text_encoder(inputs)
     assert outputs[ENCODER_OUTPUT].shape[1:] == text_encoder.output_shape
+
+
+def test_hf_auto_transformer_use_pretrained():
+    """This test ensures that use_pretrained is always True when using the auto_transformer text encoder even if a
+    user explicitly sets it to False."""
+    config = {
+        "input_features": [
+            text_feature(
+                encoder={
+                    "type": "auto_transformer",
+                    "use_pretrained": False,
```

Comment:
Ideally this should be an error if we were more strict with our config validation rules. We should instead just leave this out of the config.